src/healthkit.cs

```diff
@@ -2442,4 +2557,220 @@ interface HKWorkoutRouteQuery {
 		IntPtr Constructor (HKWorkoutRoute workoutRoute, HKWorkoutRouteBuilderDataHandler dataHandler);
 	}
 
+	[Watch (5,0), iOS (12,0)]
+	[BaseType (typeof(NSObject))]
+	[DisableDefaultCtor]
+	interface HKWorkoutBuilder
+	{
+		[NullAllowed, Export ("device", ArgumentSemantic.Copy)]
+		HKDevice Device { get; }
+
+		[NullAllowed, Export ("startDate", ArgumentSemantic.Copy)]
+		NSDate StartDate { get; }
+
+		[NullAllowed, Export ("endDate", ArgumentSemantic.Copy)]
+		NSDate EndDate { get; }
+
+		[Export ("workoutConfiguration", ArgumentSemantic.Copy)]
+		HKWorkoutConfiguration WorkoutConfiguration { get; }
+
+		[Internal]
+		[Export ("metadata", ArgumentSemantic.Copy)]
+		NSDictionary _Metadata { get; }
+
+		[Wrap ("_Metadata")]
+		HKMetadata Metadata { get; }
+
+		[Export ("workoutEvents", ArgumentSemantic.Copy)]
+		HKWorkoutEvent[] WorkoutEvents { get; }
+
+		[Export ("initWithHealthStore:configuration:device:")]
+		IntPtr Constructor (HKHealthStore healthStore, HKWorkoutConfiguration configuration, [NullAllowed] HKDevice device);
+
+		[Async]
+		[Export ("beginCollectionWithStartDate:completion:")]
+		void BeginCollection (NSDate startDate, Action<bool, NSError> completionHandler);
+
+		[Async]
+		[Export ("addSamples:completion:")]
+		void Add (HKSample[] samples, Action<bool, NSError> completionHandler);
+
+		[Async]
+		[Export ("addWorkoutEvents:completion:")]
+		void Add (HKWorkoutEvent[] workoutEvents, Action<bool, NSError> completionHandler);
+
+		[Async, Internal]
+		[Export ("addMetadata:completion:")]
+		void Add (NSDictionary metadata, Action<bool, NSError> completionHandler);
+
+		[Async]
+		[Wrap ("Add (metadata.Dictionary, completionHandler)")]
+		void Add (HKMetadata metadata, Action<bool, NSError> completionHandler);
+
+		[Async]
+		[Export ("endCollectionWithEndDate:completion:")]
+		void EndCollection (NSDate endDate, Action<bool, NSError> completionHandler);
+
+		[Async]
+		[Export ("finishWorkoutWithCompletion:")]
+		void FinishWorkout (Action<HKWorkout, NSError> completionHandler);
+
+		[Export ("discardWorkout")]
+		void DiscardWorkout ();
+
+		[Export ("elapsedTimeAtDate:")]
+		double ElapsedTimeAtDate (NSDate date);
+
+		[Export ("statisticsForType:")]
+		[return: NullAllowed]
+		HKStatistics Statistics (HKQuantityType quantityType);
+
+		[Export ("seriesBuilderForType:")]
+		HKSeriesBuilder SeriesBuilder (HKSeriesType seriesType);
+	}
+
+	delegate void HKQuantitySeriesSampleQueryQuantityDelegate (HKQuantitySeriesSampleQuery query, HKQuantity quantity, NSDate date, bool done, NSError error);
```

Comment:
just make sure that `bool done` isn't an out/ref param

Reply:
AFAIK the delegate is correct:

void (^)(HKQuantitySeriesSampleQuery *query, HKQuantity *quantity, NSDate *date, BOOL done, NSError *error))quantityHandler;